apps/x509.c

```diff
@@ -156,6 +159,12 @@ const OPTIONS x509_options[] = {
      "Clears all the prohibited or rejected uses of the certificate"},
     {"badsig", OPT_BADSIG, '-', "Corrupt last byte of certificate signature (for test)"},
     {"", OPT_MD, '-', "Any supported digest"},
+#ifndef OPENSSL_NO_SM2
+    {"sm2-id", OPT_SM2ID, 's',
+     "Specify an ID string to verify an SM2 certificate request"},
+    {"sm2-hex-id", OPT_SM2HEXID, 's',
+     "Specify a hex ID string to verify an SM2 certificate request"},
+#endif
```

Comment:
There is `-sigopt` already, no need to make the SM2 id special

Reply:
Actually using sm-id would make code easier reading, if only use sigopt, then have to resolve sm2-id from string "sm2id:...", it seems that openssl does't use this way

Reply:
The SM2 EVP_PKEY_METHOD recognises `sm2_id`.  You are using it in the test recipe, so you know this.